src/resource/resource.internal.ts

```diff
@@ -164,9 +164,11 @@ function copyNonClassProperties(source: object): object {
       copy[key] = value;
       return;
     }
-    if (value.constructor.name !== "Object") {
+
+    if (value.constructor?.name !== "Object") {
```

Comment:
@Vinnl - "**_is that why the branch coverage is <100 now_**" - yes, exactly. The reason I needed to add the ` ?.` operator was that N3 uses `Object.create(null)` [here](https://github.com/rdf-ext/rdf-dataset-indexed/blob/master/lib/N3Store.js#L17), and this code was blowing up on that (as `value.constructor` was null for the `this._graphs` member variable of the N3Store instance (which is the `rdf-dataset-indexed` implementation of RDF/JS Dataset).
So I only changed the minimum relevant to the PR itself, but in the process made this function more robust I think (in other words, yes, that `else` part can be reached, and I added a specific test for that to get the coverage back to 100% (in `resource.test.ts`, test named "clones an object containing an object with no 'constructor' at all")).
Your comments about reworking the function itself could certainly be done in a separate PR to this though I guess.

Reply:
So this is a nice example why having to rely on heuristics here is so brittle, because apparently the heuristic doesn't apply to Graphy. We can widen it a bit to apply to Graphy too, but I think that means we'll have to do the exact opposite of what you did here, i.e. if the property does not have a constructor at all, assume that it came from Graphy and not copy it over. So something like this:

```suggestion
    if (typeof value.constructor === "undefined" || value.constructor.name !== "Object") {
```

Reply:
Done - 4b3f92f6